IDE project fatal error

After sometime working with Quarkus project in IDE it crashed and now project not loading.
IDE stacktrace
```
java.lang.RuntimeException: java.lang.NullPointerException
	at com.intellij.openapi.application.impl.LaterInvocator.invokeAndWait(LaterInvocator.java:157)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:450)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:455)
	at com.intellij.openapi.project.impl.ProjectManagerExImplKt.openProject(ProjectManagerExImpl.kt:458)
	at com.intellij.openapi.project.impl.ProjectManagerExImplKt.access$openProject(ProjectManagerExImpl.kt:1)
	at com.intellij.openapi.project.impl.ProjectManagerExImpl$doOpenAsync$1.invoke(ProjectManagerExImpl.kt:143)
	at com.intellij.openapi.project.impl.ProjectManagerExImpl$doOpenAsync$1.invoke(ProjectManagerExImpl.kt:58)
	at com.intellij.openapi.project.impl.ProjectUiFrameAllocator$run$progressRunner$1.apply(ProjectFrameAllocator.kt:94)
	at com.intellij.openapi.project.impl.ProjectUiFrameAllocator$run$progressRunner$1.apply(ProjectFrameAllocator.kt:71)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:244)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:188)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:624)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:698)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:646)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:623)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:175)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:244)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NullPointerException
	at com.redhat.devtools.intellij.quarkus.gradle.AbstractGradleToolDelegate.getModuleDirPath(AbstractGradleToolDelegate.java:124)
	at com.redhat.devtools.intellij.quarkus.gradle.AbstractGradleToolDelegate.scriptExists(AbstractGradleToolDelegate.java:68)
	at com.redhat.devtools.intellij.quarkus.gradle.AbstractGradleToolDelegate.isValid(AbstractGradleToolDelegate.java:75)
	at com.redhat.devtools.intellij.quarkus.tool.ToolDelegate.getDelegate(ToolDelegate.java:67)
	at com.redhat.devtools.intellij.quarkus.QuarkusModuleUtil.ensureQuarkusLibrary(QuarkusModuleUtil.java:79)
	at com.redhat.devtools.intellij.quarkus.search.QuarkusModuleComponent.moduleAdded(QuarkusModuleComponent.java:46)
	at com.intellij.openapi.module.impl.ModuleImpl.lambda$moduleAdded$2(ModuleImpl.java:252)
	at com.intellij.serviceContainer.ComponentManagerImpl.processInitializedComponents(ComponentManagerImpl.kt:1298)
	at com.intellij.openapi.module.impl.ModuleImpl.moduleAdded(ModuleImpl.java:250)
	at com.intellij.workspaceModel.ide.impl.legacyBridge.module.ModuleManagerComponentBridge$fireModuleAddedInWriteAction$1.run(ModuleManagerComponentBridge.kt:286)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:947)
	at com.intellij.workspaceModel.ide.impl.legacyBridge.module.ModuleManagerComponentBridge.fireModuleAddedInWriteAction(ModuleManagerComponentBridge.kt:284)
	at com.intellij.workspaceModel.ide.impl.legacyBridge.module.ModuleManagerComponentBridge.fireModulesAdded(ModuleManagerComponentBridge.kt:279)
	at com.intellij.workspaceModel.ide.impl.legacyBridge.module.ModuleManagerComponentBridge.access$fireModulesAdded(ModuleManagerComponentBridge.kt:45)
	at com.intellij.workspaceModel.ide.impl.legacyBridge.module.ModuleManagerComponentBridge$1.projectOpened(ModuleManagerComponentBridge.kt:56)
	at com.intellij.util.messages.impl.MessageBusImpl.invokeMethod(MessageBusImpl.java:645)
	at com.intellij.util.messages.impl.MessageBusImpl.invokeListener(MessageBusImpl.java:624)
	at com.intellij.util.messages.impl.MessageBusImpl.access$300(MessageBusImpl.java:33)
	at com.intellij.util.messages.impl.MessageBusImpl$MessagePublisher.executeOrAddToQueue(MessageBusImpl.java:208)
	at com.intellij.util.messages.impl.CompositeMessageBus$ToDirectChildrenMessagePublisher.publish(CompositeMessageBus.java:125)
	at com.intellij.util.messages.impl.MessageBusImpl$MessagePublisher.invoke(MessageBusImpl.java:166)
	at com.sun.proxy.$Proxy108.projectOpened(Unknown Source)
	at com.intellij.openapi.project.impl.ProjectManagerExImplKt$openProject$1.run(ProjectManagerExImpl.kt:470)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:214)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:196)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:805)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeAndWait$7(ApplicationImpl.java:450)
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:134)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:82)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:131)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:47)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:187)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:891)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:760)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:447)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:818)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:446)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:805)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:492)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

--- a/src/main/java/com/redhat/devtools/intellij/quarkus/QuarkusModuleUtil.java
+++ b/src/main/java/com/redhat/devtools/intellij/quarkus/QuarkusModuleUtil.java
@@ -28,11 +28,11 @@ import com.intellij.openapi.roots.RootPolicy;
 import com.intellij.openapi.roots.impl.libraries.LibraryEx;
 import com.intellij.openapi.roots.libraries.Library;
 import com.intellij.openapi.roots.libraries.LibraryTable;
-import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VfsUtil;
 import com.intellij.openapi.vfs.VfsUtilCore;
 import com.intellij.openapi.vfs.VirtualFile;
-import com.redhat.devtools.intellij.quarkus.facet.QuarkusFacet;
 import com.redhat.devtools.intellij.lsp4mp4ij.psi.internal.core.ls.PsiUtilsLSImpl;
+import com.redhat.devtools.intellij.quarkus.facet.QuarkusFacet;
 import com.redhat.devtools.intellij.quarkus.search.QuarkusModuleComponent;
 import com.redhat.devtools.intellij.quarkus.tool.ToolDelegate;
 import org.jetbrains.annotations.NotNull;
@@ -202,7 +202,7 @@ public class QuarkusModuleUtil {
         if (roots.length > 0) {
             return roots[0];
         } else {
-            return LocalFileSystem.getInstance().findFileByPath(new File(module.getModuleFilePath()).getParent());
+            return VfsUtil.findFileByIoFile(new File(module.getModuleFilePath()).getParentFile(), true);
         }
     }
 }

--- a/src/main/java/com/redhat/devtools/intellij/quarkus/gradle/AbstractGradleToolDelegate.java
+++ b/src/main/java/com/redhat/devtools/intellij/quarkus/gradle/AbstractGradleToolDelegate.java
@@ -72,8 +72,11 @@ public abstract class AbstractGradleToolDelegate implements ToolDelegate {
 
     private boolean scriptExists(Module module) {
         String path = getModuleDirPath(module);
-        File script = new File(path, getScriptName());
-        return script.exists();
+        if (path != null) {
+            File script = new File(path, getScriptName());
+            return script.exists();
+        }
+        return false;
     }
 
     @Override
@@ -110,47 +113,62 @@ public abstract class AbstractGradleToolDelegate implements ToolDelegate {
      */
     private void processDownload(Module module, Set<String> deploymentIds, List<VirtualFile>[] result) throws IOException {
         Path outputPath = Files.createTempFile(null, ".txt");
-        Path customPath = generateCustomGradleBuild(getModuleDirPath(module), outputPath, deploymentIds);
-        collectDependencies(module, customPath, outputPath, result);
-        Files.delete(outputPath);
-        Files.delete(customPath);
-    }
+        Path customBuildFile = generateCustomGradleBuild(getModuleDirPath(module), outputPath, deploymentIds);
+        Path customSettingsFile = generateCustomGradleSettings(getModuleDirPath(module), customBuildFile);
+        try {
+            collectDependencies(module, customBuildFile, customSettingsFile, outputPath, result);
+        } catch (IOException e) {
+            LOGGER.warn(e.getLocalizedMessage(), e);
+        } finally {
+            Files.delete(outputPath);
+            Files.delete(customBuildFile);
+            Files.delete(customSettingsFile);
+        }
+   }
 
-    @NotNull
+
     private String getModuleDirPath(Module module) {
+        VirtualFile dir = QuarkusModuleUtil.getModuleDirPath(module);
+        if (dir != null) {
+            dir.refresh(false, false);
+        }
+        VirtualFile script = dir!=null?dir.findChild(getScriptName()):null;
+        if (script != null && script.exists()) {
+            return dir.getPath();
+        }
         ModuleGrouper grouper = ModuleGrouper.instanceFor(module.getProject());
         List<String> names = grouper.getGroupPath(module);
         if (!names.isEmpty()) {
             ModuleManager manager = ModuleManager.getInstance(module.getProject());
             Module parentModule = manager.findModuleByName(names.get(0));
             if (parentModule != null) {
-                return QuarkusModuleUtil.getModuleDirPath(parentModule).getPath();
+                return getModuleDirPath(parentModule);
             }
         }
-        return QuarkusModuleUtil.getModuleDirPath(module).getPath();
+        return null;
     }
 
     /**
      * Collect all deployment JARs and dependencies through a Gradle specific task.
      *
      * @param module the module to analyze
-     * @param customPath the custom Gradle build file with the specific task
+     * @param customBuildFile the custom Gradle build file with the specific task
      * @param outputPath the file where the result of the specific task is stored
      * @param result the list where to place results to
      * @throws IOException if an error occurs running Gradle
      */
-    private void collectDependencies(Module module, Path customPath, Path outputPath, List<VirtualFile>[] result) throws IOException {
+    private void collectDependencies(Module module, Path customBuildFile, Path customSettingsFile, Path outputPath, List<VirtualFile>[] result) throws IOException {
         try {
             final ExternalSystemFacadeManager manager = ServiceManager.getService(ExternalSystemFacadeManager.class);
 
             ExternalSystemExecutionSettings settings = ExternalSystemApiUtil.getExecutionSettings(module.getProject(),
-                    module.getProject().getBasePath(),
+                    getModuleDirPath(module),
                     GradleConstants.SYSTEM_ID);
 
             RemoteExternalSystemFacade facade = manager.getFacade(module.getProject(), getModuleDirPath(module), GradleConstants.SYSTEM_ID);
 
             RemoteExternalSystemTaskManager taskManager = facade.getTaskManager();
-            final List<String> arguments = Arrays.asList("-b", customPath.toString(), "-q", "--console", "plain");
+            final List<String> arguments = Arrays.asList("-c", customSettingsFile.toString(), "-q", "--console", "plain");
             settings
                     .withArguments(arguments);
             ExternalSystemTaskId taskId = ExternalSystemTaskId.create(GradleConstants.SYSTEM_ID, ExternalSystemTaskType.EXECUTE_TASK, module.getProject());
@@ -179,6 +197,8 @@ public abstract class AbstractGradleToolDelegate implements ToolDelegate {
     }
 
     abstract String getScriptName();
+
+    abstract String getSettingsScriptName();
 
     abstract String getScriptExtension();
 
@@ -210,6 +230,22 @@ public abstract class AbstractGradleToolDelegate implements ToolDelegate {
             }
         }
     }
+
+    private Path generateCustomGradleSettings(String basePath, Path customBuildFile) throws IOException {
+        Path base = Paths.get(basePath);
+        Path path = base.resolve(getSettingsScriptName());
+        String content = "";
+        try (Reader reader = Files.newBufferedReader(path, StandardCharsets.UTF_8)) {
+            content = IOUtils.toString(reader);
+        } catch (IOException e) {}
+        content += System.lineSeparator() + "rootProject.buildFileName ='" + customBuildFile.getFileName().toFile().getName() + "'";
+        Path customPath = Files.createTempFile(base, null, getScriptExtension());
+        try (Writer writer = Files.newBufferedWriter(customPath, StandardCharsets.UTF_8)) {
+            IOUtils.write(content, writer);
+            return customPath;
+        }
+    }
+
 
 
     /**

--- a/src/main/java/com/redhat/devtools/intellij/quarkus/gradle/GradleGroovyToolDelegate.java
+++ b/src/main/java/com/redhat/devtools/intellij/quarkus/gradle/GradleGroovyToolDelegate.java
@@ -40,11 +40,18 @@ public class GradleGroovyToolDelegate extends AbstractGradleToolDelegate {
                     "        }" + System.lineSeparator() +
                     "    }" + System.lineSeparator() +
                     "}";
-    
+
+    @Override
     String getScriptName() {
         return "build.gradle";
     }
 
+    @Override
+    String getSettingsScriptName() {
+        return "settings.gradle";
+    }
+
+    @Override
     String getScriptExtension() {
         return ".gradle";
     }

--- a/src/main/java/com/redhat/devtools/intellij/quarkus/gradle/GradleKotlinToolDelegate.java
+++ b/src/main/java/com/redhat/devtools/intellij/quarkus/gradle/GradleKotlinToolDelegate.java
@@ -42,10 +42,17 @@ public class GradleKotlinToolDelegate extends AbstractGradleToolDelegate {
             "    writer.close()" + System.lineSeparator() +
             "}";
 
+    @Override
     String getScriptName() {
         return "build.gradle.kts";
     }
 
+    @Override
+    String getSettingsScriptName() {
+        return "settings.gradle.kts";
+    }
+
+    @Override
     String getScriptExtension() {
         return ".gradle.kts";
     }

